### ISSUE TITLE
[java] UnnecessaryImport false positive for static imports

### ISSUE THREAD

**Issue by ben-manes** [pmd/pmd#4878]
**Affects PMD Version:** 
7.0.0

**Rule:**
[UnnecessaryImport](https://docs.pmd-code.org/pmd-doc-7.0.0/pmd_rules_java_codestyle.html#unnecessaryimport)

**Description:**
When upgrading to 7.0.0 from rc4, this rule now fails for static imports.

**Code Sample demonstrating the issue:**

```java
import static com.github.benmanes.caffeine.cache.Specifications.BOUNDED_LOCAL_CACHE;
...
superClass = BOUNDED_LOCAL_CACHE;
```

**Expected outcome:**

PMD reports a violation at line [LocalCacheFactoryGenerator L18](https://github.com/ben-manes/caffeine/blob/2760df1666780cf2ef6b9c0f676d5c3eeb6ace88/caffeine/src/javaPoet/java/com/github/benmanes/caffeine/cache/LocalCacheFactoryGenerator.java#L18), but that's wrong. That's a false positive.

**Running PMD through:** *Gradle*

**Comment by ben-manes** [comment]
Oh, I was also upgrading from Gradle 8.7-rc-3 to 8.7 and I think when I stopped it mid-way it corrupted the daemon's class cache from its incremental compiler. Even though I clean and disabled caching, it still has remnants from a canceled build and that likely confused pmd. When I stopped the daemon and reran, it recompiled without any errors.
